File: src/main/java/com/atlaspharmacy/atlaspharmacy/pharmacy/domain/Pharmacy.java
package com.atlaspharmacy.atlaspharmacy.pharmacy.domain;

import com.atlaspharmacy.atlaspharmacy.generalities.domain.Address;
import com.atlaspharmacy.atlaspharmacy.users.domain.valueobjects.AverageGrade;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;


import javax.persistence.*;

@Entity
@Table(name = "pharmacies")
public class Pharmacy  {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private String description;


    private double averageGrade;
    private String email;
    private Long telephone;

    @OneToOne(cascade = CascadeType.MERGE,fetch = FetchType.EAGER)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private Address address;


    
    public Pharmacy(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }


    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
    public Pharmacy(){}

    public Pharmacy(Long id, String name, String description, String email,
                    Long telephone,Address address,Double averageGrade) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.email = email;
        this.telephone = telephone;
        this.address = address;
        this.averageGrade = averageGrade;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Long getTelephone() {
        return telephone;
    }

    public void setTelephone(Long telephone) {
        this.telephone = telephone;
    }

    public double getAverageGrade() {
        return averageGrade;
    }

    public void setAverageGrade(double averageGrade) {
        this.averageGrade = averageGrade;
    }


}

File: frontend/src/app/model/pharmacy/pharmacy.ts

import {Address} from "../address/address";
import {AverageGrade} from "@app/model/users/averageGrade";

export class Pharmacy{
  public id : Number;
  public name: String;
  public description:String;
  public email : String;
  public telephone : Number;
  public address: Address;
  public averageGrade: AverageGrade;
  checked?: boolean;
  public averageGradeCount : Number;

  constructor(id:Number, name : String,description:String,address: Address, average_grade: AverageGrade, email : String, telephone : Number){
    this.id=id;
    this.address=address;
    this.description=description;
    this.name = name;
    this.email = email;
    this.telephone=telephone;
    this.averageGrade=average_grade;
  }
}
